File: src/main/hyyscala/com/hyy/spark/scala/core/TopN.scala
package com.hyy.spark.scala.core

import org.apache.spark.{SparkConf, SparkContext}

/**
  * Created by root on 2016/12/1.
  */
object TopN {
  def main(args: Array[String]) {
    val conf = new SparkConf().setAppName("TopN").setMaster("local")
    val sc = new SparkContext(conf)
    val linesRdd=sc.textFile("textdata/top.txt")
    val pairsRdd=linesRdd.map(value=>(Integer.valueOf(value),value))
    val results=pairsRdd.sortByKey(false).map((_._2)).take(3)
    results.foreach(println(_))
  }
}

File: src/main/studyjava/com/hyy/study/core/TopN.java
package com.hyy.study.core;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.api.java.function.PairFunction;
import scala.Tuple2;

import java.util.List;

public class TopN {

	public static void main(String[] args) {
		SparkConf conf = new SparkConf().setAppName("").setMaster("local");
		JavaSparkContext sc = new JavaSparkContext(conf);
		
		JavaRDD<String> lines = sc.textFile("textdata/top.txt");
		JavaPairRDD<Integer, String> pairs = lines.mapToPair(new PairFunction<String, Integer, String>() {

			private static final long serialVersionUID = 1L;

			@Override
			public Tuple2<Integer, String> call(String value) throws Exception {
				return new Tuple2<Integer,String>(Integer.valueOf(value),value);
			}
		});
		List<String> results = pairs.sortByKey(false).map(new Function<Tuple2<Integer,String>, String>() {

			private static final long serialVersionUID = 1L;

			@Override
			public String call(Tuple2<Integer, String> tuple) throws Exception {
				return tuple._2;
			}
		}).take(3);
		for(String result: results){
			System.out.println(result);
		}
		
		sc.close();
	}
}
